docs/changes/table/13306.bugfix.rst

```diff
@@ -0,0 +1 @@
+Tables with columns with structured data can now be properly stacked.
```

Comment:
This applies to `join` and `dstack` as well. Not sure if these need explicit tests since they call the same code, but it might be a good thing.
```suggestion
Tables with columns with structured data can now be properly stacked and joined.
```

Reply:
Yes, makes sense -- implemented. I also added tests for `dstack` and `join`.